scripts/test/binaryenjs.py

```diff
@@ -41,15 +41,22 @@ def test_binaryen_js():
         f = open('a.js', 'w')
         # avoid stdout/stderr ordering issues in some js shells - use just stdout
         f.write('''
-            console.warn = function(x) { console.log(x) };
+            console.warn = console.error = console.log;
         ''')
-        binaryen_js = open(shared.BINARYEN_JS).read()
+        binaryen_js = open(which).read()
         f.write(binaryen_js)
-        if shared.NODEJS:
-            f.write(support.node_test_glue())
         test_path = os.path.join(shared.options.binaryen_test, 'binaryen.js', s)
         test_src = open(test_path).read()
+        # wrap test code with common boilerplate
+        f.write('''
+            binaryen.ready.then(function() {
+                function assert(x) { if (!x) throw Error('Test assertion failed'); }
+                // Test code goes here
+        ''')
```

Comment:
It is definitely beneficial to have this boilerplate in one place. A downside though is that our testcases are no longer complete examples of working code.

Do we have good documentation and code samples of using `ready.then`? If so then maybe it's ok to make this change.

Reply:
I see your point, yeah. Now one could also argue that these haven't been complete examples due to missing `var Binaryen = require("binaryen");" respectively `<script src="./binaryen.js"></script>` anyway. Wdyt about actually using `require("binaryen")` and replacing that with an IIFE of the contents of the tested file, instead of just concatenating? So the tests would look like normal code using the npm package, even though that's not the case.

Reply:
Regarding documentation on `ready.then`: I plan to add this to the npm package README, and we should probably sync that with the wiki here again I guess. Having two places is somewhat odd, though.

Reply:
Sounds good, thanks.

Yeah, I'm not that happy about having two places with docs. Can we pick one to be the canonical one?